hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -141,7 +145,7 @@ object HoodieSparkSqlWriter {
           .setPartitionFields(partitionColumns)
           .setPopulateMetaFields(populateMetaFields)
           .setRecordKeyFields(hoodieConfig.getString(RECORDKEY_FIELD))
-          .setKeyGeneratorClassProp(hoodieConfig.getString(KEYGENERATOR_CLASS_NAME))
+          .setKeyGeneratorClassProp(HoodieWriterUtils.getRealKeyGenerator(hoodieConfig))
```

Comment:
what does get `Real` means?

Reply:
For Spark-SQL, we assign `SqlKeyGenerator` to `hoodie.datasource.write.keygenerator.class`, and use `hoodie.sql.origin.keygen.class` to point the real user-defined key generator, for example `ComplexKeyGenerator`. When persist it  to hoodie.properties, we need to convert to the `real`one in spark-sql scene.

Reply:
Guess @leesf was suggesting "real" sounds too vague. What about `getUserProvidedKeygenerator()`

Reply:
ok, `getOriginKeyGenerator` is the chosen one.